sharding/nodesCoordinator/indexHashedNodesCoordinator_test.go

```diff
@@ -2761,3 +2761,66 @@ func TestIndexHashedGroupSelector_GetWaitingEpochsLeftForPublicKey(t *testing.T)
 		require.Equal(t, uint32(3), epochsLeft)
 	})
 }
+
+func TestIndexHashedNodesCoordinatorComputeNumberOfShards(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should fail with less than 2", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createArguments()
+		ihnc, _ := NewIndexHashedNodesCoordinator(arguments)
+
+		eligibleMap := make(map[uint32][]Validator)
+		eligibleMap[0] = []Validator{&validator{}}
+
+		nodesConfig := &epochNodesConfig{
+			eligibleMap: eligibleMap,
+		}
+		nbShards, err := ihnc.ComputeNumberOfShards(nodesConfig)
+
+		require.NotNil(t, err)
+		require.Equal(t, uint32(0), nbShards)
+	})
+
+	t.Run("should work with 2", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createArguments()
+		ihnc, _ := NewIndexHashedNodesCoordinator(arguments)
+
+		eligibleMap := make(map[uint32][]Validator)
+		eligibleMap[0] = []Validator{&validator{}}
+		eligibleMap[1] = []Validator{&validator{}}
+
+		nodesConfig := &epochNodesConfig{
+			eligibleMap: eligibleMap,
+		}
+		nbShards, err := ihnc.ComputeNumberOfShards(nodesConfig)
```

Comment:
Would it make sense to actually call this `ComputeNumberOfShardsWithoutMeta` ?

Reply:
yes and no :) 
We know right now that the `ComputeNumberOfShards` computes it without meta, but it would be better to call it `ComputeNumberOfShardsWithoutMeta`. 
For a sovereign chain calling `ComputeNumberOfShardsWithoutMeta` would make little sense because it should not know about meta. 
I would leave it like this for now.